src/main/kotlin/org/dashj/platform/dpp/util/Cbor.kt

```diff
@@ -369,28 +268,28 @@ object Cbor {
                     else -> throw IllegalArgumentException("Unknown simple datatype")
                 }
             }
-            else -> map[key] = value.toString() // ?
+            else -> throw java.lang.IllegalArgumentException(value.toString()) // the type is not known
         }
     }
 
     private fun addValueFromCborArray(value: DataItem, array: ArrayList<Any?>) {
         when (value) {
             is UnicodeString -> array.add(value.string)
-            is co.nstant.`in`.cbor.model.Number -> {
-                if (value.value.toLong() < Int.MAX_VALUE) {
+            is NegativeInteger -> {
+                if (value.value.toLong() >= Int.MIN_VALUE) {
                     array.add(value.value.toLong().toInt())
                 } else {
                     array.add(value.value.toLong())
                 }
             }
-            is HalfPrecisionFloat -> {
-                if (value.value.toLong() > Int.MIN_VALUE) {
-                    array.add(value.value.toInt())
+            is UnsignedInteger -> {
+                if (value.value.toLong() <= Int.MAX_VALUE) {
+                    array.add(value.value.toLong().toInt())
                 } else {
                     array.add(value.value.toLong())
                 }
             }
-            is NegativeInteger -> array.add(value)
+            is HalfPrecisionFloat -> array.add(value.value)
```

Comment:
`HalfPrecisionFloat` (or `float`) was originally rendered as an integer -- now will be rendered as a `float`.

This code does distinguish `Double` and `Float`, which may not be necessary.  `Double` will be the default for literals.